### PR TITLE
Electron list metadata fix

### PR DESCRIPTION
**Pull request by poojithurao**
- [x] I have added the tests to cover my changes.
- [x] I have updated the documentation and CHANGELOG accordingly.
- [x] I have read the CONTRIBUTING document.

### CODE REVIEW

File: tests/covalent_tests/workflow/electron_test.py [hunk 1]
```diff
@@ -117,3 +117,32 @@ def workflow(x):
...
+
+    workflow.build_graph([2])
+    
+    task_metadata = workflow.transport_graph.get_node_value(0, "metadata")
+    e_list_metadata = workflow.transport_graph.get_node_value(1, "metadata")
+
+    assert list(e_list_metadata["call_before"]) == []
+    assert list(e_list_metadata["call_after"]) == []
+    
+    assert e_list_metadata["executor"] == task_metadata["executor"]
```
Comment (cjao) [inline tests/covalent_tests/workflow/electron_test.py:148]: To make the test a bit more comprehensive, could you set an executor instance for `task`, like
```python3
ct.electron(executor=LocalExecutor())
def task(x, y=0):
```

**Review by cjao — CHANGES_REQUESTED** [review]
This looks good. I'm just requesting a small enhancement to your test.

File: tests/covalent_tests/workflow/electron_test.py (hunk 1 above)
Comment (poojithurao) [reply tests/covalent_tests/workflow/electron_test.py:148]: Done.

**Review by cjao — DISMISSED** [review]
Looks good, thanks for the fix!

**Comment by scottwn** [comment]
pre-commit.ci autofix

**Comment by scottwn** [comment]
pre-commit.ci autofix

**Review by scottwn — APPROVED** [review]

**Comment by cjao** [comment]
Follow up observation -- stripping the call `call_before` and `call_after` fields would also remove pip deps. This might be inconvenient if decoding the list elements requires extra pip packages to be installed in the executor.

**Comment by cjao** [comment]
Would it be better to just strip the file transfer calldeps?

**Pull request merged**